src/pages/verify/e-ids/finnish-trust-network.mdx

```diff
@@ -16,38 +16,267 @@ The requirements are described in <a href="https://www.kyberturvallisuuskeskus.f
 
-**4. Statically configured JWK sets for signing and encryption**
+where the only two parameters directly included in the URL are:
 
-You can add your static JWKS (JSON Web Key Set) by going to your Criipto Application > `OpenID Connect` > `Client JWKS`. More information on this requirement is coming soon.
+- `client_id`: contains the Client ID of your Criipto application
+- `request`: contains Request Object (the JWT whose claims hold the JSON-encoded authorization request parameters)
 
-</Highlight>
+The code sample below demonstrates creating a Request Object JWT, then building authorize URL in a Node.js/Express application.
+
+```javascript
+const { SignJWT } = require('jose');
+
+// Generate a signed JWT (Request Object)
+const createSignedRequestObject = async (signingKey) => {
+  // Define parameters for your authorization request
+  const requestObjectPayload = {
+    response_mode: 'query',
+    response_type: 'code',
+    client_id: CLIENT_ID,
+    redirect_uri: REDIRECT_URI,
+    state: state,
+    scope: 'openid',
+  };
+
+  // Create and sign the JWT
+  const signedJwt = await new SignJWT(requestObjectPayload)
+    .setProtectedHeader({
+      alg: 'RS256',
+      kid: SIGNING_KEY_KID,
+    })
+    .setIssuer(CLIENT_ID)
+    .setAudience(`https://${CRIIPTO_DOMAIN}`)
+    .sign(signingKey);
+
+  return signedJwt;
+};
+
+// Example usage in an Express route
+app.get('/login', async (req, res) => {
+  try {
+    const privateSigningKey = 'YOUR_SIGNING_KEY_HERE';
+    const requestObject = await createSignedRequestObject(privateSigningKey);
+
+    // Build authorize URL
+    const authorizeUrl = `https://${CRIIPTO_DOMAIN}/oauth2/authorize?client_id=${CLIENT_ID}&request=${requestObject}`;
+
+    res.redirect(authorizeUrl);
+  } catch (error) {
+    console.error('Error during login:', error);
+    res.status(500).send('Login failed.');
+  }
+});
+```
+
+### Encrypted token and userinfo responses
+
+Criipto will apply [JSON Web Encryption (JWE)](/verify/getting-started/glossary/#json-web-encryption-jwe) to secure `token` and `userinfo` responses resulting from FTN authentication.
+This means that after a successful [code-for-token exchange](/verify/getting-started/glossary/#code-exchange), the `id_token` (and `userinfo` if requested) will be sent to your application in a JWE format, rather than a standard [JWT](/verify/getting-started/glossary/#json-web-token-jwt).
+These JWEs are encrypted using your application's public encryption key, requiring you to use the corresponding private key to decrypt them.
+
+For a deeper dive into JWE, please refer to our blog posts:
+
+- [An Introduction to JSON Web Encryption](https://www.criipto.com/blog/json-web-encryption-introduction)
+- [Securing Authentication Flows with JSON Web Encryption](https://www.criipto.com/blog/jwe-encrypted-responses)
+
+To configure your application to handle encrypted `token` and `userinfo` responses, follow the steps below:
+
+**1. Generate a new RSA key pair**
```

Comment:
just "new key pair"; does not have to be RSA necessarily, we support EC variants

Reply:
Should we mention somewhere that we support RSA and EC? (and which algorithms?) @mickhansen

Reply:
It's described in OpenID Metadata: https://samples.criipto.io/.well-known/openid-configuration